tree/20_cfe_basics/ncf_lib.cf

```diff
@@ -249,6 +249,176 @@ archlinux::
+    package_version_equal_command => "$(redhat_knowledge.rpm_compare_equal)";
+
+  debian::
+    package_changes => "bulk";
+    package_list_command => "$(debian_knowledge.call_dpkg) -l";
+    package_list_name_regex    => ".i\s+([^\s:]+).*";
+    package_list_version_regex => ".i\s+[^\s]+\s+([^\s]+).*";
+    package_installed_regex => ".i.*"; # packages that have been uninstalled may be listed
+    package_name_convention => "$(name)=$(version)";
+    package_list_update_ifelapsed => "240";		# 4 hours
+
+    # make correct version comparisons
+    package_version_less_command => "$(debian_knowledge.dpkg_compare_less)";
+    package_version_equal_command => "$(debian_knowledge.dpkg_compare_equal)";
+
+
+  debian.have_aptitude::
+    package_add_command => "$(debian_knowledge.call_aptitude) $(debian_knowledge.dpkg_options) --assume-yes install";
+    package_list_update_command => "/usr/bin/aptitude update";
+    package_delete_command => "$(debian_knowledge.call_aptitude) $(debian_knowledge.dpkg_options) --assume-yes remove";
+    package_update_command =>  "$(debian_knowledge.call_aptitude) $(debian_knowledge.dpkg_options) --assume-yes install";
+    package_patch_command =>  "$(debian_knowledge.call_aptitude) $(debian_knowledge.dpkg_options) --assume-yes install";
+    package_verify_command =>  "/usr/bin/aptitude show";
+    package_noverify_regex => "(State: not installed|E: Unable to locate package .*)";
+
+    package_patch_list_command => "/usr/bin/aptitude --assume-yes --simulate --verbose full-upgrade";
+    package_patch_name_regex => "^Inst\s+(\S+)\s+.*";
+    package_patch_version_regex => "^Inst\s+\S+\s+\[?\(?([^\],\s]+).*";
+
+  debian.!have_aptitude::
+    package_add_command => "$(debian_knowledge.call_apt_get) $(debian_knowledge.dpkg_options) --yes install";
+    package_list_update_command => "$(debian_knowledge.call_apt_get) update";
+    package_delete_command => "$(debian_knowledge.call_apt_get) $(debian_knowledge.dpkg_options) --yes remove";
+    package_update_command =>  "$(debian_knowledge.call_apt_get) $(debian_knowledge.dpkg_options) --yes install";
+    package_patch_command =>  "$(debian_knowledge.call_apt_get) $(debian_knowledge.dpkg_options) --yes install";
+    package_verify_command => "$(debian_knowledge.call_dpkg) -s";
+    package_noverify_returncode => "1";
+
+    package_patch_list_command => "$(debian_knowledge.call_apt_get) --just-print dist-upgrade";
+    package_patch_name_regex => "^Inst\s+(\S+)\s+.*";
+    package_patch_version_regex => "^Inst\s+\S+\s+\[?\(?([^\],\s]+).*";
+
+  freebsd::
+    package_changes => "individual";
+    package_list_command => "/usr/sbin/pkg_info";
+    package_list_name_regex    => "([^\s]+)-.*";
+    package_list_version_regex => "[^\s]+-([^\s]+).*";
+    package_name_regex    => "([^\s]+)-.*";
+    package_version_regex => "[^\s]+-([^\s]+).*";
+    package_installed_regex => ".*";
+    package_name_convention => "$(name)-$(version)";
```

Comment:
yes, and in the technique too (but I guess this is because the dash does not break anything when $(version) is empty, contrary to the equal sign...)